plugins/discourse-presence/assets/javascripts/discourse/services/composer-presence-manager.js

```diff
@@ -2,63 +2,45 @@ import { cancel, debounce } from "@ember/runloop";
 import Service, { service } from "@ember/service";
 import { isTesting } from "discourse-common/config/environment";
 
-const PRESENCE_CHANNEL_PREFIX = "/discourse-presence";
-const KEEP_ALIVE_DURATION_SECONDS = 10;
+const KEEP_ALIVE = 10 * 1000; // 10 seconds
 
 export default class ComposerPresenceManager extends Service {
   @service presence;
 
-  willDestroy() {
-    this.leave();
-  }
```

Comment:
the idea is that components using this manager are calling the leave on their `willDestroy` anyway?

Reply:
I added a `debugger` in there and it was never called. Even though I navigated everywhere in the app so I decided to remove it 🤷‍♂️

Reply:
yeah, it wouldn't trigger in the normal app usage - it's for the test env where services are destroyed with the application instance between tests, but as long the service doesn't leak state outside itself/app instance there's no need for special cleanup